src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -63,6 +63,9 @@ public String getType() {
     @Column(name = "allow_online_editor", table = "programming_exercise_details")
     private Boolean allowOnlineEditor;
 
+    @Column(name = "allow_online_editor_read_only_mode", table = "programming_exercise_details")
+    private Boolean allowOnlineEditorReadOnlyMode;
```

Comment:
```suggestion
    private boolean allowOnlineEditorReadOnlyMode = false;
```
New `boolean` attributes should be non-nullable if possible.

(related open issue #6079)